### PR TITLE
allow configuration of rsyslog port to something else than 514

### PR DESCRIPTION
**Pull request by strus38**
#396

### CODE REVIEW

**Review by btravouillon — CHANGES_REQUESTED** [review]
Thanks for the proposal, this is almost good. You must change the template rsyslog.conf.j2 to use the new variable instead of 514.

Why is there a merge commit in this PR?

File: roles/core/log_server/defaults/main.yml
Comment (btravouillon) [inline roles/core/log_server/defaults/main.yml:2]:
```suggestion
```

Comment (strus38) [reply roles/core/log_server/defaults/main.yml]: fixed... btw, you can squash my commits while merging if needed.

**Comment by strus38** [comment]
the merge could be the rebase I did on the bluebanquise master to resync my forked repo.

**Review by btravouillon — DISMISSED** [review]
Looks good to me.

**Comment by oxedions** [comment]
Thank you for this PR.
On the server side, it is ok for me.

Can you also add something on the client side ? In this PR or in another after this one.
Line `*.* @@{{networks[j2_node_main_network]['services_ip']['log_ip']}}:514` in the template need to handle different port also.

If you prefer to do it in another PR, please inform me so I merge this one.

**Comment by strus38** [comment]
Of course, I will in the same... Will do that this afternoon.

**Comment by strus38** [comment]
Done with client part too. please @oxedions let's go for a review.

File: roles/core/log_client/templates/rsyslog.conf.j2
Comment (oxedions) [inline roles/core/log_client/templates/rsyslog.conf.j2:19]: Sorry, not here 🙂 
You should add this variable to line 93 of the file. See my precedent post.
This part here is only a comment (see the # at beginning of line), that could even be removed to clean the template (this is a relic of devs).

File: roles/core/log_client/defaults/main.yml
Comment (oxedions) [inline roles/core/log_client/defaults/main.yml:2]: Maybe call it log_client_rsyslog_server_port ? To explain this is the server port, and not the client port ?

**Review by oxedions — CHANGES_REQUESTED** [review]

File: roles/core/log_client/defaults/main.yml
Comment (strus38) [reply roles/core/log_client/defaults/main.yml]: well done... :-) ... did not see the last line of the template.
It is all fixed ... hopefully.

**Review by oxedions — DISMISSED** [review]
Ok for me 😊 

Thanks a lot !

File: roles/core/log_client/readme.rst
Comment (btravouillon) [inline roles/core/log_client/readme.rst:22]: No big deal, but what was the change for 1.0.2? Should we add it as well?

**Review by btravouillon — COMMENTED** [review]
While I understand the proposal, I find it disturbing to have 2 variables to set the server port in the stack.

Use cases:

1. I want to use default port
    - I don't set any variable in my inventory
    - Server listens on 514
    - Client connects to server:514

2. I want to use a custom port
    - I set `log_server_rsyslog_port` to 5514 in my inventory
    - Server listens on 5514
    - Client connects to server:5514

3. I want to connect to an rsyslog server external to the cluster and use default port
    - I set `network.j2_node_main_network.services_ip.log_ip` to x.y.z.a
    - Client connects to x.y.z.a:514

4. I want to connect to an rsyslog server external to the cluster and use a custom port
    - I set `network.j2_node_main_network.services_ip.log_ip` to x.y.z.a in my invetory
    - I set `log_client_rsyslog_server_port` to 5514 in my inventory
    - Client connects to x.y.z.a:5514

The current proposal does not address use case 2. Having to define both `log_server_rsyslog_port` and `log_client_rsyslog_server_port` with the same value is somewhat redundant.

Any thought?

**Comment by btravouillon** [comment]
Another thing I missed yesterday evening while reviewing: the task firewalld open the default port. If one define a custom port, we have to add it to the firewall rules.

**Comment by strus38** [comment]
Well, it can also be useful to have 2 variables since it may add more
customization choices, with a server and a client connected to another
server for test purpose or something else.

Le sam. 3 oct. 2020 à 18:42, Bruno Travouillon <notifications@github.com> a
écrit :

**Comment by btravouillon** [comment]
Yes, 2 variables are useful for use case 4. I wonder if we can do better to support use case 2 with a single variable. Something similar to `log_client_rsyslog_server_port | default(log_server_rsyslog_port) | default(514)`.

**Comment by strus38** [comment]
If you don't mind, let it like that for now, and we can always open another
pr later.

Le dim. 4 oct. 2020 à 00:57, Bruno Travouillon <notifications@github.com> a
écrit :

**Review by btravouillon — CHANGES_REQUESTED** [review]
Test is failing, the client (management2) cannot connect to the server (management1):
```
[root@management2 ~]# systemctl status rsyslog
● rsyslog.service - System Logging Service
   Loaded: loaded (/usr/lib/systemd/system/rsyslog.service; enabled; vendor preset: enabled)
   Active: active (running) since Sun 2020-10-04 14:15:23 UTC; 26min ago
     Docs: man:rsyslogd(8)
           http://www.rsyslog.com/doc/
 Main PID: 5498 (rsyslogd)
    Tasks: 3 (limit: 12531)
   Memory: 1.4M
   CGroup: /system.slice/rsyslog.service
           └─5498 /usr/sbin/rsyslogd -n

Oct 04 14:15:23 management2 systemd[1]: Started System Logging Service.
Oct 04 14:15:23 management2 rsyslogd[5498]: cannot connect to 10.10.0.1:5514: Permission denied [v8.37.0-13.el8 try http://www.rsyslog.com/e/2027 ]
Oct 04 14:15:23 management2 rsyslogd[5498]: action 'action-7-builtin:omfwd' suspended (module 'builtin:omfwd'), retry 0. There should be messages before this>
Oct 04 14:15:23 management2 rsyslogd[5498]: cannot connect to 10.10.0.1:5514: Permission denied [v8.37.0-13.el8 try http://www.rsyslog.com/e/2027 ]
Oct 04 14:15:23 management2 rsyslogd[5498]: action 'action-7-builtin:omfwd' suspended (module 'builtin:omfwd'), next retry is Sun Oct  4 14:15:53 2020, retry>
Oct 04 14:16:24 management2 rsyslogd[5498]: cannot connect to 10.10.0.1:5514: Permission denied [v8.37.0-13.el8 try http://www.rsyslog.com/e/2027 ]
```

Inventory:
```
log_server_rsyslog_port: 5514
log_client_rsyslog_server_port: 5514
```

Set SELinux to permissive:
```
[root@management2 ~]# getenforce 
Enforcing
[root@management2 ~]# setenforce permissive
[root@management2 ~]# getenforce 
Permissive
```

```
[root@management2 ~]# systemctl restart rsyslog
[root@management2 ~]# systemctl status rsyslog
● rsyslog.service - System Logging Service
   Loaded: loaded (/usr/lib/systemd/system/rsyslog.service; enabled; vendor preset: enabled)
   Active: active (running) since Sun 2020-10-04 14:41:35 UTC; 986ms ago
     Docs: man:rsyslogd(8)
           http://www.rsyslog.com/doc/
 Main PID: 5601 (rsyslogd)
    Tasks: 3 (limit: 12531)
   Memory: 1.2M
   CGroup: /system.slice/rsyslog.service
           └─5601 /usr/sbin/rsyslogd -n

Oct 04 14:41:35 management2 rsyslogd[5601]: environment variable TZ is not set, auto correcting this to TZ=/etc/localtime  [v8.37.0-13.el8 try http://www.rsy>
Oct 04 14:41:35 management2 systemd[1]: Started System Logging Service.
Oct 04 14:41:35 management2 rsyslogd[5601]:  [origin software="rsyslogd" swVersion="8.37.0-13.el8" x-pid="5601" x-info="http://www.rsyslog.com"] start
Oct 04 14:41:35 management2 rsyslogd[5601]: cannot connect to 10.10.0.1:5514: No route to host [v8.37.0-13.el8 try http://www.rsyslog.com/e/2027 ]
Oct 04 14:41:35 management2 rsyslogd[5601]: action 'action-7-builtin:omfwd' suspended (module 'builtin:omfwd'), retry 0. There should be messages before this>
Oct 04 14:41:35 management2 rsyslogd[5601]: cannot connect to 10.10.0.1:5514: No route to host [v8.37.0-13.el8 try http://www.rsyslog.com/e/2027 ]
Oct 04 14:41:35 management2 rsyslogd[5601]: action 'action-7-builtin:omfwd' suspended (module 'builtin:omfwd'), next retry is Sun Oct  4 14:42:05 2020, retry>
Oct 04 14:41:35 management2 rsyslogd[5601]: cannot connect to 10.10.0.1:5514: No route to host [v8.37.0-13.el8 try http://www.rsyslog.com/e/2027 ]
Oct 04 14:41:35 management2 rsyslogd[5601]: action 'action-7-builtin:omfwd' suspended (module 'builtin:omfwd'), retry 1. There should be messages before this>
Oct 04 14:41:35 management2 rsyslogd[5601]: cannot connect to 10.10.0.1:5514: No route to host [v8.37.0-13.el8 try http://www.rsyslog.com/e/2027 ]
```

It eventually succeeds when I stop firewalld on management1.

**Comment by strus38** [comment]
I didn't update the firewalk code .. that will be one more variable.

Le dim. 4 oct. 2020 à 16:46, Bruno Travouillon <notifications@github.com> a
écrit :

**Comment by btravouillon** [comment]
The SELinux permission denial on the client:
```
type=AVC msg=audit(1601820923.322:1539): avc:  denied  { name_connect } for  pid=5498 comm=72733A6D61696E20513A526567 dest=5514 scontext=system_u:system_r:syslogd_t:s0 tcontext=system_u:object_r:unreserved_port_t:s0 tclass=tcp_socket permissive=0
type=AVC msg=audit(1601820923.322:1540): avc:  denied  { name_connect } for  pid=5498 comm=72733A6D61696E20513A526567 dest=5514 scontext=system_u:system_r:syslogd_t:s0 tcontext=system_u:object_r:unreserved_port_t:s0 tclass=tcp_socket permissive=0
```

**Comment by oxedions** [comment]
> While I understand the proposal, I find it disturbing to have 2 variables to set the server port in the stack.
@btravouillon 

Same here. I was wondering if solution is not just to do like with time or slurm roles. Merge both server and client, since both use the same package/service, with the same template, slightly different.
Then, only need a variable to defined if roles is to be used as **server** or **client** to be defined in the playbook (cf time/slurm), and a single variable `log_server_port`, that can then be used for both server and client.

This would cover all scenarios.

We can do that in another PR.

**Comment by strus38** [comment]
So, could we summarize?
- do we merge this one like this? Or should I open the firewall before
merging it?
- create another one to merge server/client role?

If so, I can work on that.

**Comment by btravouillon** [comment]
It can't merge as-is because the feature is broken when firewall is enabled or selinux is enforced.

> Merge both server and client, since both use the same package/service, with the same template, slightly different.

Yes, that could be a solution. However, it would bring new breaking changes... It would need a deprecation period (introduce new role `log` and keep legacy roles `log_server` and `log_client`).

@strus38 If I'm correct your immediate need is to define an alternate server port in the role log_client to connect rsyslog to another syslog server not configured with role log_server, which is close to use case 4. If so, focus on this role and drop the changes in log_server for now. Use variable `log_client_server_port` (i.e. prefix the variable `server_port` with the role name).

**Comment by strus38** [comment]
well, let me try to fix it correctly on both server and client (without mergin both roles). They may use the other role, or they will anyway.

File: roles/core/log_client/tasks/main.yml
Comment (btravouillon) [inline roles/core/log_client/tasks/main.yml:41]: To use `seport`, we need to install some additional packages on the target. You can refer to #394 for examples.
Otherwise, the role fails:
```
TASK [log_client : Allow syslog port into SELinux] *****************************
Sunday 04 October 2020  20:53:04 -0400 (0:00:01.024)       0:00:04.104 ******** 
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ModuleNotFoundError: No module named 'seobject'
failed: [management2] (item=tcp) => changed=false 
  ansible_loop_var: item
  item: tcp
  msg: Failed to import the required Python library (policycoreutils-python) on management2's Python /usr/libexec/platform-python. Please read module documentation and install in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ModuleNotFoundError: No module named 'seobject'
failed: [management2] (item=udp) => changed=false 
  ansible_loop_var: item
  item: udp
  msg: Failed to import the required Python library (policycoreutils-python) on management2's Python /usr/libexec/platform-python. Please read module documentation and install in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter
```

Comment (strus38) [reply roles/core/log_client/tasks/main.yml:41]: on the management1 it seems to work:
```
[root@management1 bluebanquise]# service rsyslog status
Redirecting to /bin/systemctl status rsyslog.service
● rsyslog.service - System Logging Service
   Loaded: loaded (/usr/lib/systemd/system/rsyslog.service; enabled; vendor preset: enabled)
   Active: active (running) since Mon 2020-10-05 15:25:53 CEST; 5min ago
     Docs: man:rsyslogd(8)
           https://www.rsyslog.com/doc/
 Main PID: 76054 (rsyslogd)
    Tasks: 9 (limit: 11479)
   Memory: 1.7M
   CGroup: /system.slice/rsyslog.service
           └─76054 /usr/sbin/rsyslogd -n

Oct 05 15:25:53 management1 systemd[1]: Starting System Logging Service...
Oct 05 15:25:53 management1 rsyslogd[76054]: [origin software="rsyslogd" swVersion="8.1911.0-3.el8" x-pid="76054" x-in>
Oct 05 15:25:53 management1 systemd[1]: Started System Logging Service.
Oct 05 15:25:53 management1 rsyslogd[76054]: imjournal: journal files changed, reloading...  [v8.1911.0-3.el8 try http>
[root@management1 bluebanquise]# netstat -lntp | grep 514
tcp        0      0 0.0.0.0:10514           0.0.0.0:*               LISTEN      76054/rsyslogd
tcp6       0      0 :::10514                :::*                    LISTEN      76054/rsyslogd
[root@management1 bluebanquise]# netstat -lnup | grep 514
udp        0      0 0.0.0.0:10514           0.0.0.0:*                           76054/rsyslogd
udp6       0      0 :::10514                :::*                                76054/rsyslogd
```

and on the computes:
```
[root@management1 bluebanquise]# ssh root@c001
Last login: Mon Oct  5 14:50:23 2020
[root@c001 ~]# service rsyslog status
Redirecting to /bin/systemctl status rsyslog.service
● rsyslog.service - System Logging Service
   Loaded: loaded (/usr/lib/systemd/system/rsyslog.service; enabled; vendor preset: enabled)
   Active: active (running) since Mon 2020-10-05 15:51:56 CEST; 4min 7s ago
     Docs: man:rsyslogd(8)
           https://www.rsyslog.com/doc/
 Main PID: 11410 (rsyslogd)
    Tasks: 3 (limit: 12478)
   Memory: 1.8M
   CGroup: /system.slice/rsyslog.service
           └─11410 /usr/sbin/rsyslogd -n

Oct 05 15:51:56 c001 systemd[1]: Stopped System Logging Service.
Oct 05 15:51:56 c001 systemd[1]: Starting System Logging Service...
Oct 05 15:51:56 c001 rsyslogd[11410]: [origin software="rsyslogd" swVersion="8.1911.0-3.el8" x-pid="11410" x-info="htt>
Oct 05 15:51:56 c001 systemd[1]: Started System Logging Service.
Oct 05 15:51:56 c001 rsyslogd[11410]: imjournal: journal files changed, reloading...  [v8.1911.0-3.el8 try https://www>
```
```
cat /etc/rsyslog.conf
...
# remote host is: name/ip:port, e.g. 192.168.0.1:514, port optional
*.* @@10.10.0.1:10514
...
```

Comment (btravouillon) [reply roles/core/log_client/tasks/main.yml:41]: Is SELinux enabled on c001? Is it a diskfull install or a diskless setup?

File: roles/core/log_client/vars/RedHat_7.yml
Comment (btravouillon) [inline roles/core/log_client/vars/RedHat_7.yml:2]: The variable must be `log_client_packages_to_install`, the role log_client does not use `log_server_*` variables.
Also, note that vars/RedHat_X.yml will override vars/main.yml, so you need to add the requested package **rsyslog** here.

File: roles/core/log_client/tasks/main.yml
Comment (strus38) [reply roles/core/log_client/tasks/main.yml:41]:
```
[root@c001 ~]# sestatus
SELinux status:                 enabled
SELinuxfs mount:                /sys/fs/selinux
SELinux root directory:         /etc/selinux
Loaded policy name:             targeted
Current mode:                   enforcing
Mode from config file:          enforcing
Policy MLS status:              enabled
Policy deny_unknown status:     allowed
Memory protection checking:     actual (secure)
Max kernel policy version:      31
```

it is a diskless installation.

File: roles/core/log_client/vars/RedHat_7.yml
Comment (strus38) [reply roles/core/log_client/vars/RedHat_7.yml]: done... sorry for this fix to be so complex to fix for me :-)

File: roles/core/log_client/vars/RedHat_7.yml
Comment (btravouillon) [inline roles/core/log_client/vars/RedHat_7.yml:6]: This variable is the same than in vars/main.yml, you can remove it here.

File: roles/core/log_client/vars/RedHat_7.yml
Comment (btravouillon) [inline roles/core/log_client/vars/RedHat_7.yml:8]: This variable is the same than in vars/main.yml, you can remove it here.

File: roles/core/log_client/vars/RedHat_8.yml
Comment (btravouillon) [inline roles/core/log_client/vars/RedHat_8.yml:6]: This variable is the same than in vars/main.yml, you can remove it here.

File: roles/core/log_client/vars/RedHat_8.yml
Comment (btravouillon) [inline roles/core/log_client/vars/RedHat_8.yml:8]: This variable is the same than in vars/main.yml, you can remove it here.

**Review by btravouillon — CHANGES_REQUESTED** [review]
For RedHat 7 and 8 OS families, we only need to override `log_client_packages_to_install` to add the packages for SELinux.
There is no need to override `log_client_services_to_start` and `log_client_rsyslog_conf_path` if their values are identical to vars/main.yml.

File: roles/core/log_client/vars/RedHat_8.yml
Comment (strus38) [reply roles/core/log_client/vars/RedHat_8.yml]: fixed too :-)

File: roles/core/log_client/tasks/main.yml
Comment (btravouillon) [reply roles/core/log_client/tasks/main.yml:41]: You may have run other roles before **time** that installed the requested RPMS.

```
# clush -Lw c[001-002] grep BLUEBANQUISE_IMAGE_NAME /etc/os-release
c001: BLUEBANQUISE_IMAGE_NAME="noselinux"
c002: BLUEBANQUISE_IMAGE_NAME="selinux"
# clush -Lw c[001-002] getenforce
c001: Disabled
c002: Enforcing
# ansible-playbook playbooks/logs.yml --limit c001,c002
[WARNING]: Unable to parse /etc/bluebanquise/internal as an inventory source

PLAY [log server playbook] *************************************************************************************************************************************************************************************
skipping: no hosts matched

PLAY [log clients playbook] ************************************************************************************************************************************************************************************

TASK [Gathering Facts] *****************************************************************************************************************************************************************************************
Monday 05 October 2020  12:18:07 -0400 (0:00:00.045)       0:00:00.045 ********
ok: [c001]
ok: [c002]

TASK [log_client : Gathering OS specific variables] ************************************************************************************************************************************************************
Monday 05 October 2020  12:18:08 -0400 (0:00:01.362)       0:00:01.408 ********

TASK [log_client : Install packages] ***************************************************************************************************************************************************************************
Monday 05 October 2020  12:18:08 -0400 (0:00:00.106)       0:00:01.514 ********
ok: [c001]
ok: [c002]

TASK [log_client : Template >> /etc/rsyslog.conf] **************************************************************************************************************************************************************
Monday 05 October 2020  12:18:10 -0400 (0:00:01.413)       0:00:02.927 ********
ok: [c001]
ok: [c002]

TASK [log_client : Allow syslog port into SELinux] *************************************************************************************************************************************************************
Monday 05 October 2020  12:18:11 -0400 (0:00:00.960)       0:00:03.887 ********
skipping: [c001] => (item=tcp)
skipping: [c001] => (item=udp)
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ModuleNotFoundError: No module named 'seobject'
failed: [c002] (item=tcp) => changed=false
  ansible_loop_var: item
  item: tcp
  msg: Failed to import the required Python library (policycoreutils-python) on c002's Python /usr/libexec/platform-python. Please read module documentation and install in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ModuleNotFoundError: No module named 'seobject'
failed: [c002] (item=udp) => changed=false
  ansible_loop_var: item
  item: udp
  msg: Failed to import the required Python library (policycoreutils-python) on c002's Python /usr/libexec/platform-python. Please read module documentation and install in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter

TASK [log_client : Manage services state] **********************************************************************************************************************************************************************
Monday 05 October 2020  12:18:11 -0400 (0:00:00.712)       0:00:04.600 ********
ok: [c001] => (item=rsyslog)

PLAY RECAP *****************************************************************************************************************************************************************************************************
c001                       : ok=4    changed=0    unreachable=0    failed=0    skipped=2    rescued=0    ignored=0
c002                       : ok=3    changed=0    unreachable=0    failed=1    skipped=1    rescued=0    ignored=0
```

Your latest commits fixed this condition.

Idea (💡): For diskless, when we enable support for SELinux in the image, we should install python3-policycoreutils.

**Comment by btravouillon** [comment]
1. We can cover use case 2 with the change below:
```
diff --git a/roles/core/log_client/defaults/main.yml b/roles/core/log_client/defaults/main.yml
index cef6616..fcacff9 100644
--- a/roles/core/log_client/defaults/main.yml
+++ b/roles/core/log_client/defaults/main.yml
@@ -1,2 +1,2 @@
 ---
-log_client_rsyslog_server_port: 514
\ No newline at end of file
+log_client_rsyslog_server_port: "{{ log_server_rsyslog_port | default(514) }}"
```

2. I noticed the new task `Add custom port to firewall's foo zone` add 514/tcp to firewall, which is redundant with existing configuration.

3. @oxedions should we keep **rsyslog** in the name of these new variables? We will want to keep this new functionality if one day we decide to replace rsyslog. I believe `log_server_port` and `log_client_server_port` would make more sense to prevent breaking changes in future releases.

**Comment by strus38** [comment]
I did change the variable names removing the rsyslog.
I did implement the suggestion regarding the default value of the variable on client side
However, to be coherent, why `log_client_rsyslog_conf_path` is left, should be `log_client_conf_path`, no?

**Comment by btravouillon** [comment]
`log_client_rsyslog_conf_path` is internal to the role, it is not possible to change it in the inventory. If we might change the role later, we could change the variable name without introducing breaking changes for the existing user base.

**Comment by btravouillon** [comment]
Port `514/tcp` is added with service `rsh` in [vars/main.yml](https://github.com/bluebanquise/bluebanquise/blob/54bf50a49aceffcfeba49157e28b2001ad84c9a0/roles/core/log_server/vars/main.yml#L12). It was a lazy trick to get the port in the list. Anyway, that's not a blocker for the merge.
I'll review the latest changes asap.

**Review by btravouillon — APPROVED** [review]
Looks good to me. Thanks for this PR.

**Comment by btravouillon** [comment]
BTW, please squash before merge.

**Review by oxedions — APPROVED** [review]
Ok for me.

Thanks a lot for this PR 😊

**Comment by btravouillon** [comment]
The commit message is weird:

```
$ git log -1
commit ba1d22a0413b2e58466a2731e2067dbd6f2743ee (HEAD -> master, upstream/master, origin/master, origin/HEAD)
Author: strus38 <strus38@users.noreply.github.com>
Date:   Tue Oct 6 09:19:58 2020 +0200

    allow configuration of rsyslog port to something else than 514 (#397)

    * add log_server_rsyslog_port variable to customize rsyslogd port defaulted to UDP 514 to something else

    * add log_server_rsyslog_port variable to customize rsyslogd port defaulted to UDP 514 to something else

    * add log_server_rsyslog_port variable to customize rsyslogd port defaulted to UDP 514 to something else

    * add log_client_rsyslog_port variable to customize rsyslogd port defaulted to UDP 514 to something else

    * add log_client_rsyslog_server_port variable to customize rsyslogd port defaulted to UDP 514 to something else

    * fix firewall and selinux to authorize customer ports

    * fix missing packages for selinux

    * fixing variables names

    * fixing variables duplicates

    * fixing variables names
```

When squashing, you can remove useless commit messages. 😉

**Comment by oxedions** [comment]
Due to the squash ?

**Comment by btravouillon** [comment]
Yes, by default the squash will create a [single commit with a list of the commit messages for all of the squashed commits, in date order](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#merge-message-for-a-squash-merge). GitHub allows to change the message when squashing.

**Pull request merged**
